dev/com.ibm.ws.concurrent.persistent/src/com/ibm/ws/concurrent/persistent/db/DatabaseTaskStore.java

```diff
@@ -276,13 +279,46 @@ public boolean createProperty(String name, String value) throws Exception {
             em.persist(property);
             em.flush();
         } catch (EntityExistsException x) {
-            return false;
+            /*
+             * FIXME JPA Spec says em.persist should throw EntityExistsException if entity already exists.
+             * EclipseLink implementation throws PersistanceException during the flush operation.
+             * If EclipseLink changes behavior, this exception will notify us that we need to reactor
+             * this catch to return false, and subsequent PersistenceException can remove extra checks.
+             */
+            throw new UnsupportedOperationException("EntityExistsException thrown during persist operation. "
+                                                    + "This behavior was not previously seen, and needs to be handled by the PersistanceExecutor.", x);
         } catch (PersistenceException x) {
             // Some JPA providers may throw PersistenceException for some scenarios where it can be
             // determined the row really does exist; handle those here
+
+            //TODO Remove later - Caused by Constraint Violation
             if (x.getCause() instanceof SQLIntegrityConstraintViolationException) {
                 return false;
             }
+
+            //TODO Remove later - Caused by Constraint Violation using generic SQLException
+            DatabaseException dbException = x.getCause() instanceof DatabaseException ? (DatabaseException) x.getCause() : null;
+            if (dbException != null) {
+                SQLException sqlException = dbException.getCause() instanceof SQLException ? (SQLException) dbException.getCause() : null;
+                if (sqlException != null) {
+
+                    //Investigate SQL State
+                    String message = sqlException.getLocalizedMessage();
+                    String sqlState = sqlException.getSQLState();
+                    int errorCode = sqlException.getErrorCode();
+
+                    //SQL State 23505 - Unique Constraint Violation
+                    //Suggests that we are trying to insert a row, but the primary key already exists.
+                    if (sqlState.equalsIgnoreCase("23505")) {
```

Comment:
Postgre, Derby, and DB2 all use SQL State 23505 for unique constraint violation, which I think is specific enough that we do not need to investigate the individual error code.  Additionally, Postgre (in our tests) returns an error code of 0 - successful along with the 23505 state. 
Oracle and SQL Server use the more general SQL State 23000 for any integrity constraint violation.  Currently I only check for 23505 but we could also look into 23000 + error code.

Reply:
Completely agree.  Error codes are vendor-specific and there is no sense in consulting those (would need to know which vendor it is) when we can use the standard SQL States.  As far as we are concerned, the detection of any integrity constraint violation should be sufficient (code was previously intending to detect that via java.sql.SQLIntegrityConstraintViolationException, but did so erroneously due to the extra EclipseLink chained exceptions).  Let's make this as generic and standards-based as we possibly can, in order to have the best chance of working with all databases, even those we haven't tried out.  To accomplish that, I'd recommend a check for anything under 23xxx for constraint violations (need to confirm if this is accurate).  The other thing we can do to be even more generic is just step through every exception in the chain, regardless of EclipseLink-specific classes.  Something like this (I know it won't compile, but just to get the idea across),
```
for (Throwable cause = x.getCause(); cause != null; cause = cause.getCause())
    if (cause instanceof SQLIntegrityContstraintViolationException ||
        cause instanceof SQLException && sqlState != null && sqlState.length() == 5 && sqlState.startsWith("23")) ...
```